File: suite/Ack/Ack.rs
fn ack(m: u64, n: u64) -> u64 {
    if m == 0 {
        n + 1
    } else if n == 0 {
        ack(m - 1, 1)
    } else {
        ack(m - 1, ack(m, n - 1))
    }
}

fn main_loop(iters: u64, m: u64, n: u64) -> u64 {
    let res = ack(m, n);
    if iters == 1 {
        println!("{}", res);
        0
    } else {
        main_loop(iters - 1, m, n)
    }
}

fn main() {
    let mut args = std::env::args();
    args.next();
    let iters = args
        .next()
        .expect("Missing Argument iterations")
        .parse::<u64>()
        .expect("Iterations must be a number");
    let m = args
        .next()
        .expect("Missing Argument m")
        .parse::<u64>()
        .expect("m must be a number");
    let n = args
        .next()
        .expect("Missing Argument n")
        .parse::<u64>()
        .expect("n must be a number");
    std::process::exit(main_loop(iters, m, n) as i32)
}

File: suite/Ack/Ack.ml
let rec ack m n = 
  if m=0 then n+1 
  else if n=0 then ack (m-1) 1 
  else ack (m-1) (ack m (n-1))

let rec main_loop iters m n =
  let res = ack m n in
  if iters = 1 then print_endline (string_of_int res)
  else main_loop (iters-1) m n

let main = 
  let iters = int_of_string Sys.argv.(1) in 
  let m = int_of_string Sys.argv.(2) in 
  let n = int_of_string Sys.argv.(3) in
  main_loop iters m n 


